src/modules/vax/VaxLocationDetailDrawer.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react';
+
+import { Kuota } from '~/data/types';
+
+import { VaccinationDataWithDistance } from './types';
+
+import { ArrowBackIcon, ExternalLinkIcon } from '@chakra-ui/icons';
+import {
+  Badge,
+  Box,
+  Divider,
+  Drawer,
+  DrawerBody,
+  DrawerContent,
+  DrawerHeader,
+  DrawerOverlay,
+  Link,
+  Stack,
+  Table,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr,
+  useColorMode,
+  useColorModeValue
+} from '@chakra-ui/react';
+import { format, formatDistanceToNow, parse } from 'date-fns';
+import idLocale from 'date-fns/locale/id';
+
+export interface VaxLocationDetailDrawerProps {
+  locationData: VaccinationDataWithDistance;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export default function VaxLocationDetailDrawer({ isOpen, onClose, locationData }: VaxLocationDetailDrawerProps) {
+  const { colorMode } = useColorMode();
+  const subtextColor = useColorModeValue('gray.600', 'gray.300');
+  const kuotaAvailableTextColor = useColorModeValue('green.600', 'green.400');
+  const kuotaEmptyTextColor = useColorModeValue('red.600', 'red.400');
+
+  const {
+    nama_lokasi_vaksinasi: namaLokasi,
+    alamat_lokasi_vaksinasi: alamatLokasi,
+    wilayah,
+    kecamatan,
+    kelurahan,
+    jadwal,
+    detail_lokasi,
+    last_updated_at: lastUpdated
+  } = locationData;
+
+  const mapsUrl = detail_lokasi?.[0]
+    ? `https://www.google.com/maps/search/${encodeURIComponent(`${detail_lokasi[0].lat}, ${detail_lokasi[0].lon}`)}`
+    : `https://www.google.com/maps/search/${encodeURIComponent(namaLokasi)}`;
+  return (
+    <Drawer isOpen={isOpen} onClose={onClose} placement="left" size="md">
+      <DrawerOverlay />
+      <DrawerContent>
+        <Badge
+          fontSize="0.8em"
+          onClick={() => {
+            onClose();
+          }}
+          p={4}
+          style={{
+            cursor: 'pointer'
+          }}
+        >
+          <ArrowBackIcon h={4} w={4} />
+          Kembali
+        </Badge>
```

Comment:
This should be wrapped in a `Box` and replaced with a `Button` or `IconButton`. As mentioned earlier, **never** use `onClick` on non-interactable elements (e.g. `div`, `span`).

Reply:
ahh, Sorry for that. 
Changed it to `IconButton`. Hope it's okay.

![image](https://user-images.githubusercontent.com/3627108/125578740-93577f82-2edc-4fd6-b812-de3d92e35fa4.png)

d358d555551be3ae8ef741c04f865718b4b73005